connexion/middleware/main.py

```diff
@@ -308,6 +308,9 @@ def _build_middleware_stack(self) -> t.Tuple[ASGIApp, t.Iterable[ASGIApp]]:
                 app = middleware(app)  # type: ignore
             apps.append(app)
 
+        # We sort the APIs by base path so that the most specific APIs are registered first.
```

Comment:
There might also be an issue with the ordering of routes inside an API (when adding the operations of a spec to an API)

Reply:
Indeed, we can probably quite easily reuse the same sorting function there?

Reply:
Yes, I'll investigate further in a separate issue as I did a small test that seemed to work fine even though the routes were in the "wrong" order.